### PR TITLE
Clean up default mass/inertia values so test_M113 matches report for …

### PR DESCRIPTION
**Pull request by jcmadsen**
…validation.

Produces simulations and output results to validate against report by Hamed, Wallin & Shabana, for a motion driven M113.
clean up and fix functions for outputting data.

### CODE REVIEW

File: src/demos/trackVehicle/ModelDefs.h [hunk 1]
```diff
@@ -62,15 +62,23 @@ enum Enum {
 // relative to the chassis when facing forward.
 enum VehicleSide { RIGHTSIDE, LEFTSIDE };
 
+// specify what type of output 
+enum DebugType {
```
Comment (andrewseidl) [inline src/demos/trackVehicle/ModelDefs.h]: This enum is treated as a namespace on lines 56 and 104 of ChTrackVehicle.cpp, but is not defined as such.

File: src/demos/trackVehicle/subsys/base/ChTrackVehicle.cpp [hunk 2]
```diff
@@ -51,6 +53,7 @@ ChTrackVehicle::ChTrackVehicle(const std::string& name,
       m_stepsize(step_size),
       m_save_log_to_file(false),  // save the DebugLog() info to file? default false
       m_log_what_to_file(0),      // set this in Setup_log_to_file(), if writing to file
+      m_log_debug_type(DebugType::DBG_BODY),    // default
```
Comment (andrewseidl) [inline src/demos/trackVehicle/subsys/base/ChTrackVehicle.cpp]: Enum `DebugType` is not currently a namespace.

File: src/demos/trackVehicle/subsys/base/ChTrackVehicle.cpp [hunk 3]
```diff
@@ -99,6 +102,7 @@ ChTrackVehicle::ChTrackVehicle(ChSystem* system,
       m_stepsize(system->GetStep()),
       m_save_log_to_file(false),  // save the DebugLog() info to file? default false
       m_log_what_to_file(0),      // set this in Setup_log_to_file(), if writing to file
+      m_log_debug_type(DebugType::DBG_BODY),
```
Comment (andrewseidl) [inline src/demos/trackVehicle/subsys/base/ChTrackVehicle.cpp]: Enum `DebugType` is not currently a namespace.

File: src/demos/trackVehicle/subsys/base/ChTrackVehicle.h [hunk 4]
```diff
@@ -162,11 +155,36 @@ class CH_SUBSYS_API ChTrackVehicle : public ChShared {
...
+    }
+
+    // normal force abs. vector of the persistent contact point
+    virtual const ChVector<>& Get_SG_Persistent_Fn(int track, int idx) const {
+        return ChVector<>(); 
+    }
+
+    // abs. pos. of all shoe-gear contacts found
+    virtual const std::vector<ChVector<> >& Get_SG_PosAbs_all(int track) const {
+        return std::vector<ChVector<>>();
```
Comment (andrewseidl) [inline src/demos/trackVehicle/subsys/base/ChTrackVehicle.h]: Should be `std::vector< ChVector<> >();` (note the spaces), at least until we default to C++11.

File: src/demos/trackVehicle/subsys/base/ChTrackVehicle.h [hunk 5]
```diff
@@ -162,11 +155,36 @@ class CH_SUBSYS_API ChTrackVehicle : public ChShared {
...
+    }
+
+    // abs. normal force of all sh oe-gear contacts
+    virtual const std::vector<ChVector<> >& Get_SG_Fn_all(int track) const {
+        return std::vector<ChVector<>>();
```
Comment (andrewseidl) [inline src/demos/trackVehicle/subsys/base/ChTrackVehicle.h]: Should be `std::vector< ChVector<> >();` (note the spaces), at least until we default to C++11.

**Comment by andrewseidl** [comment]
There are also numerous warnings re: returning reference of temporary: `ChTrackVehicle.h`, lines 168, 173, 178, 183.

Please fix up, amend (`git commit -a --amend`), and force push.

**Comment by jcmadsen** [comment]
those are OK, I didn't want to make those pure virtuals, but they need to be exposed in the base class for interaction with the Irrlicht GUI

**Comment by jcmadsen** [comment]
so they return nothing by default (temporary), and are meant to be implemented by inheriting classes

**Comment by rserban** [comment]
Code such as 
```    
virtual const ChVector<>& Get_SG_Persistent_Fn(int track, int idx) const {}
```
is incorrect and does not compile (MSVC).  Your default implementation must return a value.  Of course, now you have an issue since you declared that it will return a const reference.   One solution is to change the signature to return a ChVector<> and have the default implementation return some vector.  For example:
```
   virtual ChVector<> Get_SG_Persistent_Fn(int track, int idx) const {return ChVector<>(0,0,0);}
```

**Comment by jcmadsen** [comment]
Fixed based on radu's comments. Requires implementing base abstract class for children subsystems.

**Comment by jcmadsen** [comment]
this branch somehow got behind the develop Head, so just merging and resolving conflicts.

**Comment by andrewseidl** [comment]
Closing: the changes have been included in [5018f39](https://github.com/projectchrono/chrono/commit/5018f392e0b32e74f2d84b1bc68de6ce9ae7128a).

**Pull request closed without merging**